src/shogun/machine/Machine.cpp

```diff
@@ -71,9 +71,38 @@ bool Machine::train(std::shared_ptr<Features> data)
 	return result;
 }
 
-bool Machine::train(const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab){
-	set_labels(lab);
-	return train(data);
+bool Machine::train(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)
+{
+
+	auto sub = connect_to_signal_handler();
+	bool result = false;
+
+	if (support_feature_dispatching())
+	{
+		require(data != NULL, "Features not provided!");
```

Comment:
this should be just `require(data, "Features not provided!");`, as it will use the `bool` operator of `shared_ptr`